packages/lesswrong/lib/postSideRecommendations.tsx

```diff
@@ -0,0 +1,174 @@
+  StrategySpecification,
+} from "./collections/users/recommendationSettings";
+import { useRecommendationAnalytics } from "../components/recommendations/useRecommendationsAnalytics";
+
+type RecommendablePost = PostsWithNavigation|PostsWithNavigationAndRevision;
+
+/**
+ * The content to be displayed as recommendations at the side of posts is
+ * generated in the following format, which allows easily switching between
+ * different algorithms and configurations. To add a new algorithm, simply
+ * create a React hook function of type `RecommendationsGenerator` and add
+ * it to the array of generators returned by `getAvailableGenerators`.
+ */
+export type PostSideRecommendations = {
+  /** If recommendations are still loading */
+  loading: boolean,
+  /** Title for the section (generally the algorithm name) */
+  title: string,
+  /** Whether to use a numbered list or a bullet list */
+  numbered: boolean,
+  /** List of recommendations to display - may be empty if loading */
+  items: ComponentType[],
+  /** An optional cookie name to handle hiding the section - if undefined
+   *  the section is not hideable */
+  hideCookieName?: string,
+}
+
+/**
+ * Recommendations generators are used as hooks, so the name should begin
+ * with "use".
+ */
+type RecommendationsGenerator = (post: RecommendablePost) => PostSideRecommendations;
+
+const useMoreFromTheForumRecommendations: RecommendationsGenerator = (
+  _post: RecommendablePost,
+) => {
+  // TODO: Add the correct link URLs
+  const usefulLinks = "#";
+  const podcast = "/posts/K5Snxo5EhgmwJJjR2/announcing-ea-forum-podcast-audio-narrations-of-ea-forum";
+  const digest = "https://effectivealtruism.us8.list-manage.com/subscribe?u=52b028e7f799cca137ef74763&id=7457c7ff3e";
+  const jobs = "#";
+  return {
+    loading: false,
+    title: "More from the Forum",
+    numbered: false,
+    items: [
+      () => <li>
+        New to the forum? <Link to={usefulLinks}>Useful links here</Link>
+      </li>,
+      () => <li>
+        Listen to posts on our <Link to={podcast}>podcast</Link>
+      </li>,
+      () => <li>
+        Sign up for the <Link to={digest}>EA Forum Digest</Link> to get
+        curated recommendations
+      </li>,
+      () => <li>
+        Browse <Link to={jobs}>job opportunities</Link>
+      </li>,
+    ],
+    hideCookieName: HIDE_MORE_FROM_THE_FORUM_RECOMMENDATIONS_COOKIE,
+  };
+}
+
+const LiPostRecommendation: FC<{
+  post: PostsListWithVotesAndSequence,
+}> = ({post}) => {
+  const url = postGetPageUrl(post);
+  const author = userGetDisplayName(post.user);
+  const readTimeMinutes = Math.max(post.readTimeMinutes, 1);
+  const mins = readTimeMinutes === 1 ? "1 min" : `${readTimeMinutes} mins`;
+  const {ref, onClick} = useRecommendationAnalytics<HTMLLIElement>(post._id);
+  return (
+    <li ref={ref} onClick={onClick}>
+      <Link to={url}>{post.title}</Link> ({author}, {mins})
+    </li>
+  );
+}
+
+const useGeneratorWithStrategy = (
```

Comment:
Seems slightly surprising to me that this return an items as a list of react nodes. My intuition would have been to structure it to return the posts and have `LiPostRecommendation` live inside the `PostSideRecommendations.tsx` file.